src/syntax/rules.rkt

```diff
@@ -86,15 +75,17 @@
         (append (operator-info op 'itype) (append-map loop args))]
       [_ '()]))))
 
-(define (type-of-rule input output ctx)
-  (cond   ; special case for 'if', return the 'type-of-rule' of the ift branch
+(define (repr-of-rule input output ctx)
```

Comment:
Remind me—why can't this use `repr-of`? It looks to me like `repr-of-rule` just returns the `repr-of` of either the input or output expression. Why do we need to look at both, why isn't just the output (and context) enough?

Reply:
I double checked. These are slightly different operations. I cleaned it up a bit though.